### PR TITLE
Allow 100 streams initially rather than 10.

### PR DESCRIPTION
**Pull request by nmittler**
The HTTP/2 spec suggests 100 as the recommended minimum for SETTINGS_MAX_CONCURRENT_STREAMS (https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-6.5.2).  We should use this value as our default.

### CODE REVIEW

**Comment by nmittler** [comment]
@ejona86 PTAL

File: netty/src/main/java/io/grpc/transport/netty/BufferingHttp2ConnectionEncoder.java [hunk 1]
```diff
@@ -79,12 +79,19 @@
...
   // Smallest stream id whose corresponding frames do not get buffered.
   private int largestCreatedStreamId;
   private boolean receivedSettings;
 
   protected BufferingHttp2ConnectionEncoder(Http2ConnectionEncoder delegate) {
+    this(delegate, SMALLEST_MAX_CONCURRENT_STREAMS);
+  }
+
+  protected BufferingHttp2ConnectionEncoder(Http2ConnectionEncoder delegate,
+                                            int initialMaxConcurrentStreams) {
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/BufferingHttp2ConnectionEncoder.java]: It doesn't seem we use any other value for max concurrent streams (even in tests). Is there a reason for this override?

**Comment by ejona86** [comment]
@nmittler LGTM

File: netty/src/main/java/io/grpc/transport/netty/BufferingHttp2ConnectionEncoder.java (hunk 1 above)
Comment (nmittler) [inline netty/src/main/java/io/grpc/transport/netty/BufferingHttp2ConnectionEncoder.java]: Not at the moment.  I'm anticipating that this code will eventually become part of Netty and we may want the flexibility. Adding it now doesn't add too much complexity so I figured it wouldn't hurt.

**Comment by nmittler** [comment]
Cherry-picked as 5efbb6f

**Pull request closed without merging**